src/software/replay/replay_logger.cpp

```diff
@@ -0,0 +1,69 @@
+//
+// Created by jordan on 5/2/20.
+//
+#include "software/replay/replay_logger.h"
+#include <fstream>
+
+#include "software/logger/logger.h"
+
+namespace fs = std::experimental::filesystem;
+
+ReplayLogger::ReplayLogger(const std::string& out_dir_path, int _frames_per_chunk) :
+output_dir_path(out_dir_path), current_chunk(), current_chunk_idx(0),
```

Comment:
```suggestion
output_dir_path(out_dir_path), current_chunk_idx(0),
```

@jordanzliu this stopped the weird virtual function calls and `SIGABRTs` for me, might be a lead?

Reply:
Might try instantiating `current_chunk` as `TbotsReplay::default_instance()` and see if that helps